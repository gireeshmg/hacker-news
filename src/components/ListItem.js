import React from "react";
import TimeAgo from "javascript-time-ago";
import en from "javascript-time-ago/locale/en";

const getDomainName = (url) => {
  return `(${new URL(url).hostname.replace("www.", "")})`;
};

const ListItem = (props) => {
  const { post, hideHandler, upVoteHandler } = props;
  TimeAgo.addLocale(en);
  const timeAgo = new TimeAgo();

  return (
    <tr>
      <td className="center-align-text">{post.num_comments}</td>
      <td className="center-align-text">{post.points}</td>
      <td className="center-align-text">
        <button className="btn-upvote" onClick={(e) => upVoteHandler(post.objectID)}>
          Up vote
        </button>
      </td>
      <td>
        {<a className="news-anchor" href={post.url}>{post.title}</a>}
        <span className="domain-name">{post.url && getDomainName(post.url)} by</span> <span className="author">{post.author}</span>
        <span className="time">{timeAgo.format(new Date(post.created_at).getTime())}</span>
        <button className="btn-hide" onClick={() => hideHandler(post.objectID)}>
          [ hide ]
        </button>
      </td>
    </tr>
  );
};

export default ListItem;
