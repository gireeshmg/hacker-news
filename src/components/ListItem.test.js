import React from 'react';
import { render } from '@testing-library/react';
import ListItem from './ListItem';

const post = {
  author: "js2",
  comment_text: null,
  created_at: "2020-06-07T13:44:03.000Z",
  created_at_i: 1591537443,
  num_comments: 0,
  objectID: "23447271",
  parent_id: null,
  points: 2,
  story_id: null,
  story_text: null,
  story_title: null,
  story_url: null,
  title: "Wrath of the Killdozer",
  url: "https://www.damninteresting.com/the-wrath-of-the-killdozer/"
}

const hideHandler = () => {
  console.log('hideHandler clicked');
}

const upVoteHandler = () => {
  console.log('upVoteHandler clicked');
}

test('renders List items', () => {
  const { container } = render(<table><tbody>
    <ListItem hideHandler={hideHandler} upVoteHandler={upVoteHandler} post={post} />
  </tbody>
  </table>
  );
  expect(container.firstChild).toMatchSnapshot()
});
