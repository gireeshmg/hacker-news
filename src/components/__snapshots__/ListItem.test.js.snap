// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders List items 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="center-align-text"
      >
        0
      </td>
      <td
        class="center-align-text"
      >
        2
      </td>
      <td
        class="center-align-text"
      >
        <button
          class="btn-upvote"
        >
          Up vote
        </button>
      </td>
      <td>
        <a
          class="news-anchor"
          href="https://www.damninteresting.com/the-wrath-of-the-killdozer/"
        >
          Wrath of the Killdozer
        </a>
        <span
          class="domain-name"
        >
          (damninteresting.com)
           by
        </span>
         
        <span
          class="author"
        >
          js2
        </span>
        <span
          class="time"
        >
          2 hours ago
        </span>
        <button
          class="btn-hide"
        >
          [ hide ]
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
